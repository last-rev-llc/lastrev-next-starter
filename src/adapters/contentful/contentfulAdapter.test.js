import Adapter from './contentfulAdapter';
import mockContent from './contentfulAdapter.mock';

const urlMap = {
  pageGeneral: {
    url: '/[key]',
    key: 'slug'
  }
};

const transform = Adapter({
  urlMap,
  linkContentType: 'uieCta',
  manualEntryTypeText: 'Manual URL',
  modalActionText: 'Open a modal',
  contentRefTypeText: 'ContentReference'
});

describe('Contentful Adapter', () => {
  it('converts the data correctly', () => {
    const out = transform(mockContent);
    expect(out).toMatchSnapshot();
  });
});
