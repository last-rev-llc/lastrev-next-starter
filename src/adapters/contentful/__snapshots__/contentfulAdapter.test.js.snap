// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contentful Adapter converts the data correctly 1`] = `
Object {
  "contentTypeId": "1234",
  "footerLogo": Object {
    "contentType": "image/png",
    "description": "nulla hic corporis",
    "filename": "image.png",
    "height": 70,
    "size": 231,
    "title": "Mock Image for footerLogo",
    "url": "//images.ctfassets.net/o8vbqs3spmqj/mock_settingsGlobal-footerLogo/460fedebd309366e91bbe10428654ab7/image.png",
    "width": 70,
  },
  "footerSocialLinks": Array [
    Object {
      "action": "Open in a new window",
      "as": null,
      "destinationType": "Manual URL",
      "download": false,
      "href": "https://www.linkedin.com/company/uw-health",
      "icon": "Linkedin",
      "internalTitle": "LinkedIn social media CTA",
      "isModal": false,
      "linkText": "L",
      "manualUrl": "https://www.linkedin.com/company/uw-health",
      "target": "_blank",
    },
  ],
  "footerUtilityNavCta": Object {
    "action": "Open a modal",
    "as": "/testslug",
    "contentReference": Object {
      "fields": Object {
        "body": Object {
          "en-US": "Bacon is healthy!",
        },
        "slug": "testslug",
        "title": Object {
          "en-US": "Hello, World!",
        },
      },
      "sys": Object {
        "contentType": Object {
          "sys": Object {
            "id": "pageGeneral",
            "linkType": "ContentType",
            "type": "Link",
          },
        },
        "createdAt": "2016-12-20T10:43:35.772Z",
        "id": "5KsDBWseXY6QegucYAoacS",
        "revision": 1,
        "space": Object {
          "sys": Object {
            "id": "yadj1kx9rmg0",
            "linkType": "Space",
            "type": "Link",
          },
        },
        "type": "Entry",
        "updatedAt": "2016-12-20T10:43:35.772Z",
        "version": 1,
      },
    },
    "destinationType": "ContentReference",
    "download": false,
    "href": "/[slug]",
    "internalTitle": " Mock UIE-CTA",
    "isModal": true,
    "linkText": "quis explicabo ratione",
    "target": null,
  },
  "headerMainNavLogo": Object {
    "contentType": "image/png",
    "description": "natus voluptatem voluptatibus",
    "filename": "image.png",
    "height": 60,
    "size": 219,
    "title": "Mock Image for headerMainNavLogo",
    "url": "//images.ctfassets.net/o8vbqs3spmqj/mock_settingsGlobal-headerMainNavLogo/6597774d68edd42f663c5fb16ab0f43b/image.png",
    "width": 60,
  },
  "headerSearchErrorMessage": "There was an error",
  "headerSearchPlaceholderText": "Search",
  "headerUtilityNavLinks": Array [
    Object {
      "contentTypeId": "elementNavLink",
      "id": "mock_settingsGlobal-baselineCta",
      "internalTitle": "Healthfacts - Nav Link",
      "mainLink": Object {
        "action": "Open in the same window",
        "as": null,
        "destinationType": "Manual URL",
        "download": false,
        "href": "https://melody.biz",
        "internalTitle": " Mock UIE-CTA",
        "isModal": false,
        "linkText": "quis explicabo ratione",
        "manualUrl": "https://melody.biz",
        "target": null,
        "trackingId": "4321",
      },
    },
    Object {
      "contentTypeId": "elementNavLink",
      "id": "mock_settingsGlobal-baselineMainLinks",
      "internalTitle": "Healthwise - Nav Link",
      "mainLink": Object {
        "action": "Download",
        "as": null,
        "destinationType": "Manual URL",
        "download": true,
        "href": "https://melody.biz",
        "icon": "phone",
        "internalTitle": " Mock UIE-CTA",
        "isModal": false,
        "linkText": "quis explicabo ratione",
        "manualUrl": "https://melody.biz",
        "target": null,
        "trackingId": "9838",
      },
    },
    Object {
      "contentTypeId": "elementNavLink",
      "id": "mock_Element-NavLink",
      "internalTitle": "Kids Health - Nav Link",
      "mainLink": Object {
        "action": "Open in a new window",
        "as": null,
        "assetReference": Object {
          "fields": Object {
            "file": Object {
              "contentType": "image/png",
              "details": Object {
                "image": Object {
                  "height": 250,
                  "width": 250,
                },
                "size": 12273,
              },
              "fileName": "Nyan_cat_250px_frame.png",
              "url": "//images.ctfassets.net/yadj1kx9rmg0/4gp6taAwW4CmSgumq2ekUm/9da0cd1936871b8d72343e895a00d611/Nyan_cat_250px_frame.png",
            },
            "title": "Nyan Cat",
          },
          "sys": Object {
            "createdAt": "2016-12-20T10:43:35.772Z",
            "id": "nyancat",
            "revision": 1,
            "space": Object {
              "sys": Object {
                "id": "yadj1kx9rmg0",
                "linkType": "Space",
                "type": "Link",
              },
            },
            "type": "Asset",
            "updatedAt": "2016-12-20T10:43:35.772Z",
          },
        },
        "destinationType": "Asset reference",
        "download": false,
        "href": "//images.ctfassets.net/yadj1kx9rmg0/4gp6taAwW4CmSgumq2ekUm/9da0cd1936871b8d72343e895a00d611/Nyan_cat_250px_frame.png",
        "internalTitle": " Mock UIE-CTA",
        "isModal": false,
        "linkText": "quis explicabo ratione",
        "target": "_blank",
        "trackingId": "mollitia",
      },
    },
  ],
  "headerUtilityNavSiteNameLink": Object {
    "contentTypeId": "elementNavLink",
    "id": "mock_settingsGlobal-headerUtilityNavSiteNameLink",
    "internalTitle": "Mock headerUtilityNavSiteNameLink for Settings - Global",
    "mainLink": Object {
      "action": "Open in the same window",
      "as": null,
      "destinationType": "Manual URL",
      "download": false,
      "href": "#top",
      "icon": "",
      "internalTitle": " Mock UIE-CTA",
      "isModal": false,
      "linkText": "quis explicabo ratione",
      "manualUrl": "#top",
      "style": "",
      "target": null,
      "trackingId": "mollitia",
    },
  },
  "id": "2345",
  "internalTitle": "Patient Education Global Settings",
}
`;
